oibackoff/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for oibackoff
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    (fn: Function, callback: Function): void;
```

Comment:
Hmmm... I might be overcomplicating it. Looks like the library only has full functional support for up to two arguments (error and data callback) [because of the way it handles errors](https://github.com/chilts/oibackoff/blob/master/oibackoff.js#L110). I might just limit the typing to two callback arguments for simplicity and accuracy.